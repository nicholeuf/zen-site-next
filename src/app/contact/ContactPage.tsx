import Typography from '@mui/material/Typography';

import PageContainer from '@/components/PageContainer';
import ExternalLink from '@/components/ExternalLink';
import { SMALLCHAT_ENABLED } from '@/app/lib/smallchat';

interface ContactPageProps {
  chatEnabled?: boolean;
}

const ContactPage: React.FC<ContactPageProps> = ({
  chatEnabled = SMALLCHAT_ENABLED,
}) => {
  const getCopy = () => {
    const LinkedInContent = (
      <ExternalLink
        href="https://www.linkedin.com/in/nicholeuf"
        sx={{
          fontWeight: '500',
        }}
      >
        LinkedIn
      </ExternalLink>
    );

    return (
      <Typography variant="body2">
        Please connect with me on&nbsp;{LinkedInContent}
        {chatEnabled && ' or send a message in chat'}. I look forward to hearing
        from you!
      </Typography>
    );
  };

  return (
    <PageContainer data-testid="contact-page">
      <Typography variant="h1" gutterBottom>
        Contact
      </Typography>
      {getCopy()}
    </PageContainer>
  );
};

export default ContactPage;
