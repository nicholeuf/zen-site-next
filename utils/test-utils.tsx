import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import renderer from 'react-test-renderer';

import AppLayout from '@/components/AppLayout';
import GlobalStyles from '@/app/styles/GlobalStyles';

interface Props {
  children: React.ReactNode;
}

// Include header, main, and footer in rendered content
const MockLayout: React.FC<Props> = ({ children }) => {
  return <AppLayout>{children}</AppLayout>;
};

// Include just mui theme global styles
const MockStyles: React.FC<Props> = ({ children }) => {
  return <GlobalStyles>{children}</GlobalStyles>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockStyles, ...options });

const customRenderWithLayout = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockLayout, ...options });

const customSnapshotRender = (children: ReactElement) =>
  renderer.create(<MockStyles>{children}</MockStyles>);

const customSnapshotRenderWithLayout = (children: ReactElement) =>
  renderer.create(<MockLayout>{children}</MockLayout>);

export * from '@testing-library/react';
export {
  customRender as render,
  customRenderWithLayout as renderWithLayout,
  customSnapshotRender as renderSnapshot,
  customSnapshotRenderWithLayout as renderSnapshotWithLayout,
};
