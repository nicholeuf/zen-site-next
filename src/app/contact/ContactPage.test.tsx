/*
 * @jest-environment jsdom
 */
import { renderSnapshot } from 'test-utils';

import ContactPage from './ContactPage';

describe('The Contact Page Component', () => {
  test('has expected snapshot when chat enabled', () => {
    const component = renderSnapshot(<ContactPage chatEnabled />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('has expected snapshot when chat is not enabled', () => {
    const component = renderSnapshot(<ContactPage chatEnabled={false} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
