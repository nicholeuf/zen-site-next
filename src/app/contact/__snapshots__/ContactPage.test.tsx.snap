// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Contact Page Component has expected snapshot when chat enabled 1`] = `
.emotion-0 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
  padding-bottom: 16px;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

.emotion-1 {
  margin: 0;
  font-family: fontFamily;
  font-weight: 300;
  font-size: 3.5rem;
  line-height: 1.167;
  margin-bottom: 0.35em;
}

@media (min-width:600px) {
  .emotion-1 {
    font-size: 4.7129rem;
  }
}

@media (min-width:900px) {
  .emotion-1 {
    font-size: 5.3556rem;
  }
}

@media (min-width:1200px) {
  .emotion-1 {
    font-size: 5.9983rem;
  }
}

.emotion-2 {
  margin: 0;
  font-size: 0.9rem;
  font-family: fontFamily;
  font-weight: 400;
  line-height: 1.43;
}

.emotion-3 {
  margin: 0;
  font: inherit;
  color: #FA2742;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 500;
}

.emotion-3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  className="MuiContainer-root MuiContainer-maxWidthLg emotion-0"
  data-testid="contact-page"
>
  <h1
    className="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom emotion-1"
  >
    Contact
  </h1>
  <p
    className="MuiTypography-root MuiTypography-body2 emotion-2"
  >
    Please connect with me on 
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-3"
      href="https://www.linkedin.com/in/nicholeuf"
      onBlur={[Function]}
      onFocus={[Function]}
      rel="noopener"
      target="_blank"
    >
      LinkedIn
    </a>
     or send a message in chat
    . I look forward to hearing from you!
  </p>
</div>
`;

exports[`The Contact Page Component has expected snapshot when chat is not enabled 1`] = `
.emotion-0 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
  padding-bottom: 16px;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

.emotion-1 {
  margin: 0;
  font-family: fontFamily;
  font-weight: 300;
  font-size: 3.5rem;
  line-height: 1.167;
  margin-bottom: 0.35em;
}

@media (min-width:600px) {
  .emotion-1 {
    font-size: 4.7129rem;
  }
}

@media (min-width:900px) {
  .emotion-1 {
    font-size: 5.3556rem;
  }
}

@media (min-width:1200px) {
  .emotion-1 {
    font-size: 5.9983rem;
  }
}

.emotion-2 {
  margin: 0;
  font-size: 0.9rem;
  font-family: fontFamily;
  font-weight: 400;
  line-height: 1.43;
}

.emotion-3 {
  margin: 0;
  font: inherit;
  color: #FA2742;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 500;
}

.emotion-3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  className="MuiContainer-root MuiContainer-maxWidthLg emotion-0"
  data-testid="contact-page"
>
  <h1
    className="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom emotion-1"
  >
    Contact
  </h1>
  <p
    className="MuiTypography-root MuiTypography-body2 emotion-2"
  >
    Please connect with me on 
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-3"
      href="https://www.linkedin.com/in/nicholeuf"
      onBlur={[Function]}
      onFocus={[Function]}
      rel="noopener"
      target="_blank"
    >
      LinkedIn
    </a>
    . I look forward to hearing from you!
  </p>
</div>
`;
